aten/src/ATen/mkl/SparseDescriptors.h

```diff
@@ -100,34 +100,15 @@ class MklSparseCsrDescriptor
 
     sparse_matrix_t raw_descriptor;
 
-    // Assuming that the last two dimensions are block elements of the matrix
-    if (values.dim() == 3 && crow_indices.dim() == 1 && col_indices.dim() == 1) {
```

Comment:
Instead of adding a new class, modifying this if statement to consider the layout instead of dimensions would be enough.